servicetalk-examples/grpc/debugging/src/main/java/io/servicetalk/examples/grpc/debugging/DebuggingServer.java

```diff
@@ -0,0 +1,194 @@
+ * 2022-01-06 12:48:51,628 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 - R:/127.0.0.1:55604] READ_COMPLETE
+ * 2022-01-06 12:48:51,629 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 - R:/127.0.0.1:55604] FLUSH
+ * 2022-01-06 12:48:51,629 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 - R:/127.0.0.1:55604] READ_REQUEST
+ * 2022-01-06 12:48:51,629 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 - R:/127.0.0.1:55604] USER_EVENT: io.netty.channel.socket.ChannelInputShutdownReadComplete@1a6252d6
+ * 2022-01-06 12:48:51,629 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 - R:/127.0.0.1:55604] CLOSE
+ * 2022-01-06 12:48:51,631 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 ! R:/127.0.0.1:55604] INACTIVE
+ * 2022-01-06 12:48:51,631 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0x1647dcf1, L:/127.0.0.1:8080 ! R:/127.0.0.1:55604] UNREGISTERED
+ * </pre>
+ */
+public final class DebuggingServer {
+
+    static {
+        /*
+         * 1. (optional) Disables the AsyncContext associated with individual request/responses to reduce stack-trace
+         * depth and simplify execution tracing. This will disable/break some features such as request tracing,
+         * authentication, propagated timeouts, etc. that rely upon the AsyncContext so should only be disabled when
+         * necessary for debugging:
+         */
+        // AsyncContext.disable();
+    }
+
+    public static void main(String[] args) throws Exception {
+        GrpcServers.forPort(8080)
+                .initializeHttp(builder -> {
+                    builder
+                            /*
+                             * 2. Disables most asynchronous offloading to simplify execution tracing. Changing this may
+                             * significantly change application behavior and introduce unexpected blocking. It is most
+                             * useful for being able to directly trace through situations that would normally involve a
+                             * thread handoff.
+                             */
+                            //.executionStrategy(HttpExecutionStrategies.offloadNever())
+                            /*
+                             * 3. Enables detailed logging of I/O and I/O states.
+                             * Be sure to also enable the logger in your logging config file (log4j2.xml for this example).
+                             */
+                            .enableWireLogging("servicetalk-examples-wire-logger", TRACE, Boolean.TRUE::booleanValue)
```

Comment:
For grpc (h2 based protocol) we typically don't need wire logging bcz it is a binary protocol. Would it be sufficient to just use frame logging here, and folks can reference the http example for wire logging? This would also simplify the debugging output in comments above.

Reply:
Wire logging is still useful to observe network events (connection active / closed / shutdown output). I'm agreed that the payload is not much useful. Consider using `Boolean.FALSE::booleanValue` and clarify that in the comment.